src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,310 @@
+
+    summary_df.loc[len(summary_df)] = average
+    summary_df.loc[len(summary_df)] = total_score
+    summary_df.loc[len(summary_df)] = median
+
+    summary_df.insert(0, "question index", question_indices)
+
+    response_df = result.dataset.to_pandas()
+    response_df["scores"] = scores
+
+    with pd.ExcelWriter(excel_file) as writer:
+        response_df.to_excel(writer, sheet_name="dataset", index=False)
+        summary_df.to_excel(writer, sheet_name="Summary", index=False)
+    return summary_df
+
+
+def write_results(
+    result: EvaluationResult, file_formats: str, output_dir: str, model_name: str
+):
+    response_df = result.dataset.to_pandas()
+    response_df["model_name"] = model_name
+
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    response_df["scores"] = scores
+
+    now = datetime.now()
+    timestamp = now.strftime("%m-%d-%Y_%H-%M-%S-%f")
+    response_df["evaluation_run"] = f"run-{timestamp}"
+
+    print("Responses and scores written to:")
+    for fmt in file_formats:
+        if IOFileType.JSONL.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.JSONL.value, output_dir, timestamp, model_name
+            )
+            response_df.to_json(f"{results_file}", orient="records", lines=True)
+        elif IOFileType.CSV.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.CSV.value, output_dir, timestamp, model_name
+            )
+            response_df.to_csv(f"{results_file}", index=False)
+        elif IOFileType.XLSX.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.XLSX.value, output_dir, timestamp, model_name
+            )
+            create_excel_results_file(results_file, result)
+        else:
+            logger.debug("Output format %s is not valid", fmt)
+            continue
+
+        print(f"{results_file}")
+
+
+def run_dk_bench(
+    ctx: click.Context,
+    model: str,
+    max_workers: str | int | None,
+    gpus: int | None,
+    backend: str | None,
+    enable_serving_output: bool,
+    input_questions: pathlib.Path,
+    model_prompt: str,
+    temperature: float,
+    judge_model_name: str,
+) -> tuple[EvaluationResult, str]:
+    logging.info("Running DK-Bench with settings: %s", locals())
+
+    if "OPENAI_API_KEY" not in os.environ:
+        raise EnvironmentError(
+            "Environment variable 'OPENAI_API_KEY' must be set to run the Judge model in DK-Bench."
+        )
+    judge_openai_api_key = os.environ.get("OPENAI_API_KEY", None)
+
+    # need to make sure input questions is a jsonl file
+    if input_questions.suffix.lstrip(".") != IOFileType.JSONL.value:
+        raise ValueError(
+            f"Invalid file type: {input_questions}. Expected a '.jsonl' file."
+        )
+
+    server = None
+    try:
+        evaluator = RagasEvaluator()
+        if model is not None:
```

Comment:
In your type-hinting above, `model` is hinted as only ever being set to `str`, so this check will always be True. Is there something I'm missing here?

Reply:
This check would go away per the changes I'm proposing in https://github.com/instructlab/instructlab/pull/2940/files#r1921098046

Reply:
Yeah makes sense.